geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java

```diff
@@ -68,46 +77,40 @@ public Object get(
 		return properties.get(toPropertyName(propertyName));
 	}
 
-	public void store(
+	public synchronized void store(
 			final ParameterEnum propertyName,
 			final Object value ) {
-		synchronized (MUTEX) {
-			properties.put(
-					toPropertyName(propertyName),
-					value);
-		}
+		properties.put(
+				toPropertyName(propertyName),
+				value);
 	}
 
-	public Object storeIfEmpty(
+	public synchronized Object storeIfEmpty(
 			final ParameterEnum propertyName,
 			final Object value ) {
 		final String pName = toPropertyName(propertyName);
-		synchronized (MUTEX) {
-			if (!properties.containsKey(pName)) {
-				LOGGER.info("Setting parameter : " + pName + " to " + value.toString());
-				properties.put(
-						pName,
-						value);
-				return value;
-			}
-
-			return properties.get(pName);
+		if (!properties.containsKey(pName)) {
+			LOGGER.info("Setting parameter : " + pName + " to " + value.toString());
```

Comment:
Nice!  Didn't even know you could do that.
On Apr 2, 2015 10:16 PM, "Chris Bennight" <notifications@github.com> wrote:

> In
> geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/PropertyManagement.java
> <https://github.com/ngageoint/geowave/pull/307#discussion_r27712447>:
>
> >  			final ParameterEnum propertyName,
> >  			final Object value ) {
> >  		final String pName = toPropertyName(propertyName);
> > -		synchronized (MUTEX) {
> > -			if (!properties.containsKey(pName)) {
> > -				LOGGER.info("Setting parameter : " + pName + " to " + value.toString());
> > -				properties.put(
> > -						pName,
> > -						value);
> > -				return value;
> > -			}
> > -
> > -			return properties.get(pName);
> > +		if (!properties.containsKey(pName)) {
> > +			LOGGER.info("Setting parameter : " + pName + " to " + value.toString());
>
> Instead of string concatenation, move the LOGGER statements over to use
> dynamic content injection {}
> ref: http://slf4j.org/faq.html#logging_performance
> String won't even be built if the log level is high enough.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ngageoint/geowave/pull/307/files#r27712447>.
>